### DISCUSSION TITLE
BB on Digital Ocean - Adding own domain name

### DISCUSSION THREAD

**Discussion by todenforce** [Budibase/budibase#5862] [General]
Great work with BB solution and it is everything I need to build apps.
I have got everything up and running on Digital Ocean and everything works great apart from 1 thing and would love some help.

I have a domain I want to use and I am not an expert in using any of the solutions mentioned...using Caddy or Nginx. Is there anyone that has a video of how to do this? Accessing the terminal and doing the install of these seems a bit difficult as I'm not used to this environment and don't want to mess things up so any help would be a huge benefit.

Thank you if you can help

**Comment by shogunpurple** [comment]
Hi @todenforce - thanks for the kind words. No problem, here's a video showing step by step how to set this up, where I've tried to add some context to each of the steps by typing into the terminal.

https://user-images.githubusercontent.com/11256663/168281363-d18693b8-5293-4719-a8e2-34019782454f.mp4

We do mention in the docs that those who are not comfortable with the terminal/server management would be better served by budibase cloud. What were your reasons for wanting to self host? Was it just the custom domain, or were there other reasons?

**Reply by shogunpurple** [reply]
Sorry - just realised that the bottom of the video at the end is cut off when I compressed it in iMovie. I'll re-upload 👍

**Comment by todenforce** [comment]
Hi Martin

 

I have got this far but can’t seem to get out of this section to do the last part….dont know what to press??, tried enter, cntrl enter ? im so close, also lost the last part what you did 

 

thanks for helping if you can just clarify this last bit I will be so grateful

**Reply by shogunpurple** [reply]
Hi Paul,

I can't see the image because you are using email - could you post the actual image in this comment? It seems like in general discussions has poor support for images from github emails.

Although - I have a sneaking suspicion that what you are trying to do is exit vim. There's actually a whole joke around this in the developer community, because it's not that easy.
![image](https://user-images.githubusercontent.com/11256663/168328740-d9f78d57-1968-4429-96fb-f109c45cf1c5.png)

The key combination you need to do that is:

`:wq <Enter>`

It will save the file and bring you back out to the terminal.

**Reply by todenforce** [reply]
Thanks Martin>
I got to the section where i added the details but cant see what to do next

![image](https://user-images.githubusercontent.com/12717114/168392882-4b864342-4840-4d28-a78c-134c8f5b85d5.png)
so basically, its the last part of the video which i couldnt see from here

**Comment by todenforce** [comment]
@shogunpurple thanks but not got any further im afraid, very frustrated pulling my hair out by the hour!!
I have tried using your :wq! <Enter>
whether that is supposed to be wq enter , :wq! enter, :wq enter none of them work for me

I will actually pay someone to teach me this...or for a full video with every key stroke shown at every stage in the process.

it is very very frustrating as I have clients ready to show these apps and i cant on the current domain with no https and no branded domain
my domain name is:

statim-app.solutions 

is there any reason that the name could be the problem with a hyphen in?

**Reply by shogunpurple** [reply]
@todenforce  Ok, you are nearly there

Let's try something else instead of vim. Assuming that you have `caddy` installed on your droplet, copy and paste the following commands (one by one), and then hit enter after each one.

`printf "statim-app.solutions\n\nreverse_proxy localhost:10000\n" > Caddyfile`
`ufw allow https`
`ufw allow http`
`caddy stop`
`caddy start`

**Comment by todenforce** [comment]
@shogunpurple AWESOMENESS - Excellent thank you so much, i owe you a beer or ten!!

**Comment by thiyagarajanshankar13** [comment]
[shogunpurple](https://github.com/shogunpurple) Amazing post.. This is my third day with budibase and I already love it.. And this post got me my own instance with my own domain in few mins...

Thanks Again...

**Comment by todenforce** [comment]
Thanks so much for this.The key reason is that although I am very used to the hosting environment, I have done for years, but as new things come about then l guess you have to jump on board and learn, which I will no problem.The ability to have a self hosted version is something that I have been looking for, I have had fairly bad experiences with some recent solutions that are hosted and I'm not saying you are the same by any means but I need to be able to host in certain regions due to data sovereignty and strict IT protocols.I am using Digital Ocean for my own Dev environment but I will then have to host it in certain environments with certain providers...ie Azure.Hope this makes sense it's just the nature of clients that I work with and as you know, they can be very regulated in their demands.Of course being able to access data and host the data is no problem as you have given us all the options so if they want me to access via API I can, if I need to host their data I can Inna way mentioned above.Best of luck with everything, I will be building some great tools with Buddibase and be a ln avid advocate and supporter along the journey.I work as a developer full time but I have had to use the PowerPlatform for the businesses as this is their preference. My aim is to have an alternative that meets all their IT requirements but gives them a lot more capability and flexibility with Buddibase.However, it's one step at a timeKind RegardsMany ThanksPaul
